storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -2826,9 +2915,22 @@ public void launchServer() throws Exception {
                 .parallelStream()
                 .mapToDouble(SupervisorResources::getTotalCpu)
                 .sum());
-
+            StormMetricsRegistry.registerGauge("nimbus:longest-scheduling-time-ms", () -> {
+                Long currTime = Time.nanoTime();
+                Long startTime = schedulingStartTime.get();
+                //There could be race condition here but seems trivial, elapsed is
+                // guaranteed to be no longer than real elapsed time of scheduling
+                Long longest = longestSchedulingTime.get();
+                if (startTime != null) {
+                    longest = currTime - startTime > longest ? currTime - startTime : longest;
```

Comment:
I don't understand what is happening here. Why are we potentially calculating a new longest here?

Reply:
We would like to compute the distribution of scheduler latency as well as the longest scheduling iteration. If a scheduler is stuck in the middle of a scheduling iteration, the histogram won't reflect that until the scheduling iteration has ended because timer only report the time for a complete cycle. Hence I added this gauge to track the longest scheduling iteration in real time.

Reply:
Okay, that makes sense. Thanks for explaining. Could you add a comment to this effect here?